guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,688 @@
+"""
+random_cutout = keras_cv.layers.RandomCutout(width_factor=0.4, height_factor=0.4)
+keras_cv.visualization.plot_image_gallery(
+    random_cutout(image_batch),
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+
+"""
+While this tackles the problem reasonably well, it can cause the classifier to
+develop responses to borders between features and black pixel areas caused by
+the cutout.
+
+[`CutMix`](https://arxiv.org/abs/1905.04899) solves the same issue; but by using
+a more complex (and effective!) technique.
+Instead of replacing the cut-out areas with black pixels, `CutMix` replaces
+these regions with regions of other images sampled from within your training
+set!
+Following this replacement, the image's classification label is updated to be a
+blend of the original image's class label, as well as the image that was
+overlaid into the cutout section's label.
+
+What does this look like in practice?  Let's check it out:
+"""
+cut_mix = keras_cv.layers.CutMix()
+# CutMix needs to modify both images and labels
+inputs = {"images": image_batch, "labels": label_batch}
+
+keras_cv.visualization.plot_image_gallery(
+    cut_mix(inputs)["images"],
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+"""
+Let's hold off from adding it to our augmenter for a minute - more on that
+soon!
+
+Next, let's look into `MixUp()`.
+Unfortunately, while `MixUp()` has been empirically shown to *substantially*
+improve both the robustness and the generalization of the trained model,
+it is not well-understood why such improvement occurs.
+A little alchemy never hurt anyone!
+
+![](https://i.imgur.com/d4ZZYvW.png)
+
+`MixUp()` works by sampling two images from a batch, then proceeding to
+literally blend together their pixel intensities as well as their classification
+labels.
+
+Let's see it in action:
+"""
+mix_up = keras_cv.layers.MixUp()
+# MixUp needs to modify both images and labels
+inputs = {"images": image_batch, "labels": label_batch}
+
+keras_cv.visualization.plot_image_gallery(
+    mix_up(inputs)["images"],
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+"""
+If you look closely, you'll see that the images have been blended together.
+
+Instead of applying `CutMix()` and `MixUp()` to every image, we instead pick
+one or the other to apply to each batch.
+This can be expressed using `keras_cv.layers.RandomChoice()`
+"""
+
+cut_mix_or_mix_up = keras_cv.layers.RandomChoice([cut_mix, mix_up], batchwise=True)
+augmenters += [cut_mix_or_mix_up]
+
+"""
+Applying it to your training pipeline is easy:
+"""
+
+augmenter = keras.Sequential(augmenters)
+train_ds = train_ds.map(augmenter, num_parallel_calls=tf.data.AUTOTUNE)
+
+image_batch = next(iter(train_ds.take(1)))["images"]
+keras_cv.visualization.plot_image_gallery(
+    image_batch,
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+
+"""
+We also need to resize our evaluation set, but luckily that's trivial:
```

Comment:
We also need to resize our evaluation set to get dense batches of the image size expected by our model. We use the deterministic `keras_cv.layers.Resizing` in this case to avoid adding noise to our evaluation metric.